x/config/builder.go

```diff
@@ -162,8 +162,8 @@ func buildInbound(cv *buildable, t transport.Transport, k *Kit) (transport.Inbou
 
 // buildUnaryOutbound builds an UnaryOutbound from the given value. This will panic
 // if the output type for this is not transport.UnaryOutbound.
-func buildUnaryOutbound(cv *buildable, t transport.Transport, k *Kit) (transport.UnaryOutbound, error) {
-	result, err := cv.Build(t, k)
+func buildUnaryOutbound(o *buildableOutbound, t transport.Transport, k *Kit) (transport.UnaryOutbound, error) {
+	result, err := o.Value.Build(t, k.withTransportSpec(o.TransportSpec))
```

Comment:
It's a fairly cheap copy at build time. We can optimize it later if needed. Not worth the savings at this time IMO.